gcc/rust/backend/rust-compile-base.cc

```diff
@@ -301,22 +306,34 @@ HIRCompileBase::setup_abi_options (tree fndecl, ABI abi)
     case Rust::ABI::CDECL:
       DECL_ATTRIBUTES (fndecl)
 	= tree_cons (get_identifier ("cdecl"), NULL, DECL_ATTRIBUTES (fndecl));
-      break;
+
+      return;
```

Comment:
I feel like we can make this just be handled the same as the other ABI options instead of returning.

Reply:
> I feel like we can make this just be handled the same as the other ABI options instead of returning.

Ah, sorry. I should have explained in the comments that `cdecl` is special.

We can't annotate the function with `cdecl` ABI since that's the default ABI GCC will be using, which will cause GCC to emit a warning saying the attribute is useless (which is confusing to the user as the attribute is added by us).

Reply:
Please, via a new PR, do add such rationale as a source code comment.

Reply:
> Please, via a new PR, do add such rationale as a source code comment.

Understood. Will do it in a moment.